client_admin/pulp/client/admin/tasks.py

```diff
@@ -162,6 +176,33 @@ def cancel(self, **kwargs):
             else:
                 raise
 
+    def purge(self, **kwargs):
+        """
+        Attempts to purge completed tasks with states finished, error, timed out
+        and/or skipped. If a state does not support purging, a bad request(400)
+        will be raised from the server. If no states is mentioned, it defaults to
+        purge all the completed tasks with states finished, error, timed out, skipped.
+        """
+        self.context.prompt.render_title('Purge Completed Tasks')
+
+        if kwargs.get(self.purge_all.keyword) and kwargs.get(self.purge_state.keyword):
+            msg = _('These arguments cannot be used together')
+            self.context.prompt.render_failure_message(msg)
+            return
+
+        if kwargs.get(self.purge_state.keyword):
+            states = kwargs[self.purge_state.keyword]
+        elif kwargs.get(self.purge_all.keyword):
+            states = VALID_STATES
+        else:
+            raise CommandUsage
+
+        try:
+            self.context.server.tasks.purge_tasks(state=states)
+            self.context.prompt.render_success_message(_('Task purging is successfully initiated.'))
+        except PulpServerException:
+                raise
```

Comment:
Yes. The intended behaviour is to bubble up the PulpServerException. No other additional error handling